### ISSUE TITLE
Add `ts-node` as optional peer dependency?

### ISSUE THREAD

**Issue by aaronadamsCA** [storybookjs/storybook#18774]
`start-storybook` appears to execute our `main.ts` configuration file using `ts-node` (although I'm not sure where this behaviour comes from, and [a source search turns up nothing](https://github.com/storybookjs/storybook/search?q=%22ts-node%22)).

When the only available `ts-node` is v9 (in our case due to transitive dependencies on unmaintained package `@endemolshinegroup/cosmiconfig-typescript-loader`), `start-storybook` crashes almost immediately:

```
Error: Debug Failure. False expression: Non-string value passed to `ts.resolveTypeReferenceDirective`, likely by a wrapping package working with an outdated `resolveTypeReferenceDirectives` signature. This is probably not a problem in TS itself.
```

Directly installing the latest `ts-node` beside Storybook ([in combination with this workaround](https://github.com/storybookjs/storybook/issues/16948#issuecomment-1190407954)) solves the problem.

Which leads me to ask: should Storybook define an optional peer dependency on `ts-node` v10? Because it seems to be required for TypeScript configuration file support. It might be good to indicate this version requirement in the docs, too.

**Comment by Exifers** [comment]
I have the exact same issue.
It's especially problematic with pnpm as when installing `ts-node@10` it's not visible to the dependencies, only to the package itself (unless [hoisting](https://pnpm.io/npmrc#hoist-pattern) but then it takes the wrong version, cf [here](https://github.com/pnpm/pnpm/discussions/5332)).

The only solution I found is falling back to a `main.js` file.

**Comment by IanVS** [comment]
This has been addressed in storybook 7.0.  Please try upgrading to the latest beta using `npx sb@next upgrade --prerelease`.  If you're still having troubles, let me know and we can look into it.  But you should no longer need to install `ts-node` in your project.
